autometa/taxonomy/ncbi.py

```diff
@@ -92,332 +93,305 @@ def __str__(self):
 
+            merged_df = pd.merge(
+                left=your_df,
+                right=this_df,
+                how='left',
+                left_on=<taxid_column>,
+                right_index=True)
         """
         canonical_ranks = [r for r in reversed(NCBI.CANONICAL_RANKS)]
-        canonical_ranks.remove('root')
+        canonical_ranks.remove("root")
         taxids = list(set(taxids))
-        taxids_ = {}
+        ranked_taxids = {}
         for rank in canonical_ranks:
             for taxid in taxids:
                 name = self.name(taxid, rank=rank)
-                if taxid not in taxids_:
-                    taxids_.update({taxid: {rank: name}})
+                if taxid not in ranked_taxids:
+                    ranked_taxids.update({taxid: {rank: name}})
                 else:
-                    taxids_[taxid].update({rank: name})
-        df = pd.DataFrame(taxids_).transpose()
-        df.index.name = 'taxid'
+                    ranked_taxids[taxid].update({rank: name})
+        df = pd.DataFrame(ranked_taxids).transpose()
+        df.index.name = "taxid"
         if fillna:
-            df.fillna(value='unclassified', inplace=True)
+            df.fillna(value="unclassified", inplace=True)
         return df
 
     def rank(self, taxid):
-        """Short summary.
+        """
+        Return the respective rank of provided `taxid`. If the taxid is deprecated, suppressed,
+        withdrawn from NCBI (basically old) the updated rank will be retrieved
 
         Parameters
         ----------
         taxid : int
-            `taxid` to retrieve rank from `nodes`.
+            `taxid` to retrieve rank from `nodes.dmp`
 
         Returns
         -------
         str
-            rank name
+            'rank name' if taxid is found in nodes.dmp and 'unclassified' if not
 
         Raises
-        -------
+        ------
         ValueError
-            Provided `taxid` is not an integer
-
+            Provided `taxid` is not a positive integer
         """
-        # If taxid is not found in nodes.dmp, will return 'unclassified'
-        try:
-            taxid = int(taxid)
-        except ValueError as err:
-            logger.error(
-                f'Taxid must be an integer! {taxid} type --> {type(taxid)}')
-            return None
-        return self.nodes.get(taxid, {'rank': 'unclassified'}).get('rank')
+        taxid = self.is_valid_taxid(taxid)
+        return self.nodes.get(taxid, {"rank": "unclassified"}).get("rank")
 
     def parent(self, taxid):
-        """Retrieve the parent taxid of provided taxid
+        """
+        Retrieve the parent taxid of provided `taxid`
 
         Parameters
         ----------
         taxid : int
-            Description of parameter `taxid`.
+            `taxid` whose parent `taxid` is being returned from `nodes.dmp`
 
         Returns
         -------
         int
-            parent taxid
+            Parent `taxid` if found in `nodes.dmp` otherwise 1
```

Comment:
Similar to comment in `parameters` section

```suggestion
            Parent taxid if found in `nodes.dmp` otherwise 1
```